3ds Max/Max2Babylon/Forms/AnimationGroupControl.cs

```diff
@@ -167,6 +170,40 @@ private void confirmButton_Click(object sender, EventArgs e)
             ConfirmPressed?.Invoke(confirmedInfo);
         }
 
+        private AnimationGroup CalculateEndFrameFromAnimationGroupNodes(AnimationGroup animationGroup)
+        {
+            int endFrame = 0;
+            foreach (uint nodeHandle in currentInfo.NodeHandles)
+            {
+                IINode node = Loader.Core.GetINodeByHandle(nodeHandle);
+                if (node.IsAnimated && node.TMController!=null)
+                {
+                    int lastKey = 0;
+                    if (node.TMController.PositionController != null)
+                    {
+                        int posKeys = node.TMController.PositionController.NumKeys;
+                        lastKey = Math.Max(lastKey, node.TMController.PositionController.GetKeyTime(posKeys - 1));
+                    }
+
+                    if (node.TMController.RotationController!=null)
+                    {
+                        int rotKeys = node.TMController.RotationController.NumKeys;
+                        lastKey = Math.Max(lastKey, node.TMController.RotationController.GetKeyTime(rotKeys - 1));
+                    }
+
+                    if (node.TMController.ScaleController!=null)
+                    {
+                        int scaleKeys = node.TMController.ScaleController.NumKeys;
+                        lastKey = Math.Max(lastKey, node.TMController.ScaleController.GetKeyTime(scaleKeys - 1));
+                    }
+                    decimal keyTime = Decimal.Ceiling(lastKey / 160);
+                    endFrame = Math.Max(endFrame, Decimal.ToInt32(keyTime));
+                }
+            }
+            animationGroup.FrameEnd = endFrame;
```

Comment:
Lets remove this side effect, I don't think we really want to have a "calculation" function change the state of things outside its scope, it'll be cleaner to return endFrame

Reply:
I'll move it to tools class and the user will force the calculation manually